coldfront/core/utils/views/mou_views.py

```diff
@@ -0,0 +1,144 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.views.generic.edit import FormView
+from django.views import View
+from coldfront.core.allocation.models import AllocationPeriod
+from coldfront.core.allocation.models import AllocationAdditionRequest
+from coldfront.core.allocation.models import SecureDirRequest
+from coldfront.core.project.models import SavioProjectAllocationRequest
+from coldfront.core.resource.utils_.allowance_utils.computing_allowance import ComputingAllowance
+from coldfront.core.utils.forms.file_upload_forms import PDFUploadForm
+
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from django.http import HttpResponse
+from django.http import FileResponse
+
+from flags.state import flag_enabled
+import datetime
+
+class BaseMOUView(LoginRequiredMixin, UserPassesTestMixin):
+    
+    def get_filename(self):
+        type_ = {'new-project': 'NewProject_MOU',
```

Comment:
I think you're already addressing this from our previous discussion, but noting here anyway: the types should be moved into settings to avoid hard-coding them.

Reply:
yeah done i just didn't push yet